### PR TITLE
convert string dates to time; exclude commands

### PR DESCRIPTION
**Pull request by scottlepp**

### CODE REVIEW

File: duck/duckdb.go [hunk 1]
```diff
@@ -261,3 +270,36 @@ func resultsToFrame(name string, res string, f *sdk.Frame, frames []*sdk.Frame)
...
+	".show",
+	".system",
+	".testcase",
+	".timer",
+	"read_csv",
+	"read_text",
+	"read_blob",
+	"st_read",
+	"read_json_auto",
+	"read_parquet",
```
Comment (ryantxu) [inline duck/duckdb.go:304]:
```suggestion
	"read_parquet",
	"set"
```

maybe as a regxp?

File: duck/duckdb.go [hunk 2]
```diff
@@ -278,6 +278,9 @@ func exclude(query string) (bool, string) {
 			return true, exclusion
 		}
 	}
+	// check for "set" command
+	// can't just add to exclusions since that uses Contains
+	// select * from corset - should be ok
```
Comment (samjewell) [inline duck/duckdb.go:283]: You could argue that the same applies to all the other commands too, right? like `st_read` for instance.

An LLM tells me that we could develop
> Develop a custom query parser to analyze and sanitize user input before execution:
> 1. Restrict allowed SQL keywords and functions
> 2. Validate table and column names against a predefined list
> 3. Ensure JOIN conditions use only permitted tables

I guess that's a bigger lift, but it sounded more watertight to me 🤔

File: duck/duckdb.go [hunk 3]
```diff
@@ -261,3 +270,45 @@ func resultsToFrame(name string, res string, f *sdk.Frame, frames []*sdk.Frame)
...
+	".once",
+	".open",
+	".output",
+	".parameter",
+	".read",
+	".shell",
+	".show",
+	".system",
+	".testcase",
+	".timer",
```
Comment (samjewell) [inline duck/duckdb.go:307]: How did you build this list? Wouldn't an allow-list be safer? What if future versions of DuckDB ship with more dot-commands, or something like that.

File: duck/duckdb.go (hunk 2 above)
Comment (scottlepp) [reply duck/duckdb.go:283]: yeah this is really a quick fix.  yeah `st_read` could possibly be valid.  `select * from first_reader` or something.  i will fix that

File: duck/duckdb.go (hunk 3 above)
Comment (scottlepp) [reply duck/duckdb.go:307]: all good questions.  i went through all the commands to determine what could be malicious.  if we update the duckdb version in grafana cloud, we need to revisit this.  but .  again, this is just a quick fix.  you could make the argument that the sql validation should really happen in grafana.  some users of this lib might want to use these commands.  these exclusions only happen when querying from a frame ( which probably only happens from grafana )

**Pull request closed without merging**